src/include/duckdb/function/table_function.hpp

```diff
@@ -420,6 +421,8 @@ class TableFunction : public SimpleNamedParameterFunction { // NOLINT: work-arou
 	bool sampling_pushdown;
 	//! Whether or not the table function supports late materialization
 	bool late_materialization;
+	//! Data for WITH ORDINALITY
+	ordinality_data_t ordinality_data;
```

Comment:
The table function is the wrong place for this - the TableFunction should not contain bind information. This should be stored in the bound nodes (`TableFunctionRef`/`LogicalGet`/...) instead